src/ChronosTime.php

```diff
@@ -221,11 +266,14 @@ public function setHours(int $hours): static
      */
     public function setTime(int $hours = 0, int $minutes = 0, int $seconds = 0, int $microseconds = 0): static
     {
+        $ticks = $hours * self::TICKS_PER_HOUR +
+            $minutes * self::TICKS_PER_MINUTE +
+            $seconds * self::TICKS_PER_SECOND +
+            $microseconds * self::TICKS_PER_MICROSECOND;
+        $ticks = static::mod($ticks, self::TICKS_PER_DAY);
```

Comment:
With a signed 64-bit integer we can support up to 100 million days rollover days before failing so this solution should be stable.